frontend/internal-packages/schema-bench/src/cli/evaluateWithLangsmith.ts

```diff
@@ -0,0 +1,160 @@
+  handleUnexpectedError,
+  parseArgs,
+  selectTargetDatasets,
+} from './utils/index.ts'
+
+config({ path: resolve(__dirname, '../../../../../.env') })
+
+const executorTypeSchema = v.picklist(['liamdb', 'openai'])
+const positiveIntegerSchema = v.pipe(
+  v.union([v.pipe(v.string(), v.transform(Number)), v.number()]),
+  v.integer(),
+  v.minValue(1),
+)
+const optionsSchema = v.object({
+  executorType: v.optional(executorTypeSchema, 'liamdb'),
+  numRepetitions: v.optional(positiveIntegerSchema, 3),
+  maxConcurrency: v.optional(positiveIntegerSchema, 3),
+})
+
+type ExecutorOptions = v.InferOutput<typeof optionsSchema>
+type ExecutorType = v.InferOutput<typeof executorTypeSchema>
+
+const parseExecutorAndOptions = (argv: string[]): ExecutorOptions => {
+  const args = argv.slice(2)
+
+  const rawOptions: Record<string, unknown> = {}
+
+  for (const arg of args) {
+    if (arg === '--openai') {
+      rawOptions['executorType'] = 'openai'
+    } else if (arg === '--liamdb') {
+      rawOptions['executorType'] = 'liamdb'
+    } else if (arg.startsWith('--num-repetitions=')) {
+      rawOptions['numRepetitions'] = arg.split('=')[1]
+    } else if (arg.startsWith('--max-concurrency=')) {
+      rawOptions['maxConcurrency'] = arg.split('=')[1]
+    }
+  }
+
+  return v.parse(optionsSchema, rawOptions)
+}
+
+const createTarget = (
+  executorType: ExecutorType,
+): ((input: LangSmithInput) => Promise<LangSmithOutput>) => {
+  if (executorType === 'liamdb') {
+    return async (input: LangSmithInput): Promise<LangSmithOutput> => {
+      const prompt = input.prompt || input.input || ''
+
+      const result = await executeLiamDb({ input: prompt })
+
+      if (result.isErr()) {
+        throw result.error
+      }
+
+      return { schema: result.value }
+    }
+  }
+
+  if (executorType === 'openai') {
+    const apiKey = process.env['OPENAI_API_KEY']
+    if (!apiKey) {
+      handleCliError('OPENAI_API_KEY environment variable is required')
+    }
+
+    const executor = new OpenAIExecutor({ apiKey: apiKey || '' })
+
+    return async (input: LangSmithInput): Promise<LangSmithOutput> => {
+      const prompt = input.prompt || input.input || ''
+
+      const result = await executor.execute({ input: prompt })
+
+      if (result.isErr()) {
+        throw result.error
+      }
+
+      return { schema: result.value }
+    }
+  }
+
+  return handleCliError(`Unknown executor type: ${executorType}`)
+}
+
+type ExperimentResults = Awaited<ReturnType<typeof evaluate>>
+
+const runEvaluation = (
+  datasetName: string,
+  options: ExecutorOptions,
+): ResultAsync<ExperimentResults, Error> => {
+  const target = createTarget(options.executorType)
+
+  return fromPromise(
+    evaluate(target, {
+      data: `schema-bench-${datasetName}`,
+      evaluators: [schemaEvaluator],
+      experimentPrefix: `${options.executorType}-${datasetName}`,
+      maxConcurrency: options.maxConcurrency,
+      numRepetitions: options.numRepetitions,
+    }),
```

Comment:
This is the main function for submitting evaluations to LangSmith.
ref: https://docs.langchain.com/langsmith/code-evaluator

Pass in the target and evaluators to execute and evaluate in one go. Use numRepetitions to run multiple executions.